packages/key-management/src/util/ownSignatureKeyPaths.ts

```diff
@@ -1,8 +1,64 @@
-import { AccountKeyDerivationPath, GroupedAddress, KeyRole } from '../types';
+import { AccountKeyDerivationPath, GroupedAddress } from '../types';
 import { Cardano } from '@cardano-sdk/core';
 import { isNotNil } from '@cardano-sdk/util';
 import uniq from 'lodash/uniq';
 
+type Account = {
+  rewardAccount: Cardano.RewardAccount;
+  accountDerivationPath: AccountKeyDerivationPath;
+};
+
+/**
+ * Gets whether any certificate in the provided certificate list requires a stake key signature.
+ *
+ * @param accounts The known reward accounts.
+ * @param txBody The transaction body.
+ */
+// eslint-disable-next-line complexity
+const getStakingKeyPaths = (
+  accounts: Account[],
```

Comment:
I think it would be slightly cleaner if it was mapped inside this function. Won't need the `Account` type then.
```suggestion
  addresses: GroupedAddress[],
```